src/cli_driver.rs

```diff
@@ -7,852 +7,125 @@ extern crate clap;
-                    // second pass produces ignored errors, we won't say so.
-                    tt_warning!(status, "errors were issued by the TeX engine, but were ignored; \
-                                         use --print and/or --keep-logs for details.");
-                    self.noted_tex_warnings = true;
-                }
-            },
-            Err(e) => {
-                if let Some(output) = self.io.mem.files.borrow().get(self.io.mem.stdout_key()) {
-                    tt_error!(status, "something bad happened inside TeX; its output follows:\n");
-                    tt_error_styled!(status, "===============================================================================");
-                    status.dump_to_stderr(&output);
-                    tt_error_styled!(status, "===============================================================================");
-                    tt_error_styled!(status, "");
-                }
+    sess_builder.print_stdout(args.is_present("print_stdout"));
 
-                return Err(e);
-            }
+    if let Some(items) = args.values_of_os("hide") {
+        for v in items {
+            sess_builder.hide(v);
         }
-
-        Ok(0)
     }
 
+    if let Some(p) = args.value_of("bundle") {
+        let zb = ctry!(ZipBundle::<File>::open(Path::new(&p)); "error opening bundle");
+        sess_builder.bundle(Box::new(zb));
+    } else if let Some(u) = args.value_of("web_bundle") {
+        let tb = ITarBundle::<HttpITarIoFactory>::new(&u);
+        sess_builder.bundle(Box::new(tb));
+    } else {
+        sess_builder.bundle(config.default_io_provider(status)?);
+    }
 
-    fn bibtex_pass(&mut self, status: &mut TermcolorStatusBackend) -> Result<i32> {
-        let result = {
-            let mut stack = self.io.as_stack();
-            let mut engine = BibtexEngine::new ();
-            status.note_highlighted("Running ", "BibTeX", " ...");
-            engine.process(&mut stack, &mut self.events, status,
-                           &self.tex_aux_path.to_str().unwrap())
-        };
+    let mut sess = sess_builder.create()?;
+    let result = sess.run(status);
 
-        match result {
-            Ok(TexResult::Spotless) => {},
-            Ok(TexResult::Warnings) => {
-                tt_note!(status, "warnings were issued by BibTeX; use --print and/or --keep-logs for details.");
-            },
-            Ok(TexResult::Errors) => {
-                tt_warning!(status, "errors were issued by BibTeX, but were ignored; \
-                                          use --print and/or --keep-logs for details.");
-            },
-            Err(e) => {
-                if let Some(output) = self.io.mem.files.borrow().get(self.io.mem.stdout_key()) {
-                    tt_error!(status, "something bad happened inside BibTeX; its output follows:\n");
+    if let Err(ref e) = result {
+        if let ErrorKind::EngineError(ref engine) = e.kind() {
```

Comment:
This works on the latest Rust compiler but not on the earlier version we use to do big-endian testing (cf the CircleCI failure). I think it should work to change this to:

```
if let &ErrorKind::EngineError(ref engine) = e.kind() {
```